src/tainting/Dataflow_tainting.ml

```diff
@@ -962,43 +968,71 @@ let rec check_tainted_lval env (lval : IL.lval) : Taints.t * Lval_env.t =
   report_findings { env with lval_env } findings;
   (taints, lval_env)
 
-and propagate_taint_via_unresolved_java_getters_and_setters env e args
+and propagate_taint_via_java_getters_and_setters_without_definition env e args
     all_args_taints =
   match e with
   | {
    e =
      Fetch
        ({
-          base = Var _obj;
+          base = Var obj;
           rev_offset =
-            [ { o = Dot { IL.ident = method_str, method_tok; id_info; _ }; _ } ];
+            [ { o = Dot { IL.ident = method_str, method_tok; sid; _ }; _ } ];
         } as lval);
    _;
   }
-    when env.lang =*= Lang.Java
-         && Option.is_none !(id_info.id_resolved)
-         && String.length method_str > 3 -> (
+    when env.lang =*= Lang.Java && String.length method_str > 3 -> (
       (* e.g. getFooBar/setFooBar -> fooBar *)
       let prop_str =
         String.uncapitalize_ascii (Str.string_after method_str 3)
       in
-      (* TODO: Fix this with naming info coming from Pro via some hook. *)
-      let prop_name =
+      let mk_prop_name () =
```

Comment:
it looks like `mk_prop_name` is solely used in the definition of `mk_prop_lval`. would we be able to inline it? i had to read it to understand the relationship there